spec/javascripts/components/accordion-spec.js

```diff
@@ -49,22 +63,50 @@ describe('Accordion component', function () {
 
   afterEach(function () {
     document.body.removeChild(container)
+    if (GOVUK.analytics.trackEvent.calls) {
+      GOVUK.analytics.trackEvent.calls.reset()
+    }
+  })
+
+  it('applies custom attributes to click event if specified in section', function () {
+    startAccordion()
+    document.querySelector('.govuk-accordion__section-header button').click()
+    expect(GOVUK.analytics.trackEvent).toHaveBeenCalledWith('pageElementInteraction', 'accordionOpened', { transport: 'beacon', label: 'Accordion Section Heading Text', dimension114: 1 })
+  })
+
+  it('applies custom attributes to click event if specified in show/hide section', function () {
+    var object = {
+      'track-options': {
+        dimensionExample: 'Example value'
+      }
+    }
+
+    accordion.setAttribute('data-show-all-attributes', JSON.stringify(object))
+
+    startAccordion()
+    document.querySelector('.gem-c-accordion__show-all').click()
+    expect(GOVUK.analytics.trackEvent).toHaveBeenCalledWith('pageElementInteraction', 'accordionClosed', Object({ transport: 'beacon', label: 'Hide all sections', dimensionExample: 'Example value' }))
   })
 
   it('applies given data attributes to the show/hide all link', function () {
     var object = {
       ui: {
         type: 'type value',
         section: 'section value'
+      },
+      'track-options': {
+        dimensionExample: 'Example value'
       }
     }
+
     var wrappingObject = {
       'gtm-event-name': 'example',
-      'show-all-attributes': JSON.stringify(object)
+      ...object
     }
     accordion.setAttribute('data-show-all-attributes', JSON.stringify(wrappingObject))
     startAccordion()
+    expect(document.querySelector('.govuk-accordion__show-all').getAttribute('data-track-options')).toEqual(JSON.stringify({ dimensionExample: 'Example value' }))
     expect(document.querySelector('.govuk-accordion__show-all').getAttribute('data-gtm-event-name')).toEqual('example')
-    expect(document.querySelector('.govuk-accordion__show-all').getAttribute('data-show-all-attributes')).toEqual(JSON.stringify(object))
+    expect(document.querySelector('.govuk-accordion__show-all').getAttribute('data-ui')).toEqual(JSON.stringify(object.ui))
```

Comment:
I think this is modifying the intent of the original test, which is to test that the show/hide all link has been correctly given a `data-show-all-attributes` attribute (needed for the GA4 work).

Suggestion - could this test be reverted and a new test added to check this new functionality? It feels related but distinct.